### PR TITLE
Add num logits to keep to granite llms

### PR DESCRIPTION
**Pull request by alex-jw-brooks**
This PR adds support for `num_logits_to_keep` to Granite models. This also allows Granite to be used as the language model for Llava Next models, which assume that this kwarg is part of the underlying LLM's signature. 

As an aside, should this be a requirement for the underlying LLM signature of llava next encapsulating LLMs in general? I'm happy to submit a follow-up PR to only pass it if it's part of the LLM signature if we'd like to add that flexibility 🙂 

For testing -`num_logits_to_keep` is tested under the common tests [here](https://github.com/huggingface/transformers/blob/main/tests/test_modeling_common.py#L4811); I verified that this test now runs for granite models and passes as expected.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@ArthurZucker

### CODE REVIEW

**Comment by alex-jw-brooks** [comment]
It seems this was added as part of https://github.com/huggingface/transformers/pull/35235 - will close this

**Comment by ArthurZucker** [comment]
Sorry for not reviewing earlier and thanks for contributing! 🤗
